src/main/java/hudson/plugins/git/UserRemoteConfig.java

```diff
@@ -58,6 +61,9 @@ public UserRemoteConfig(String url, String name, String refspec, @CheckForNull S
         this.name = fixEmpty(name);
         this.refspec = fixEmpty(refspec);
         this.credentialsId = fixEmpty(credentialsId);
+        if (FIPS140.useCompliantAlgorithms() && StringUtils.isNotEmpty(this.credentialsId) && StringUtils.startsWith(this.url, "http:")) {
```

Comment:
I assume that the use of the unauthenticated `git://` protocol is outside the scope of FIPS compliance, since it does not require any credential and allows read access to anyone that can access that network port on the host.

Reply:
good catch. using `git://` protocol with a credentials should be restricted in FIPS env.
done with rev 08a38cf8